### PR TITLE
Use mdbook to render doc

### PR DESCRIPTION
**Pull request by cjpatton**
This allows us to use `mdbook` to render the design doc into HTML, as suggested by @acmiyaguchi. To run, do
```
cargo install mdbook
mdbook server
```
This starts an HTTP server that updates the page each time the source is edited. This makes editing the document much a more pleasant experience, I think.

### CODE REVIEW

**Review by acmiyaguchi — COMMENTED** [review]
I suggest adding a README with the instructions to build/serve the docs. Otherwise it's a good starting point 👍.

**Review by acmiyaguchi — APPROVED** [review]

File: REAMDE.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+This document is built using Rust's
```
Comment (acmiyaguchi) [inline REAMDE.md:1]: The document is title REAMDE instead of README

Comment (cjpatton) [reply REAMDE.md:1]: Ugh, sorry.

File: book.toml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+[book]
+authors = ["Christopher Patton"]
```
Comment (tgeoghegan) [inline book.toml:2]: Not that I don't appreciate your contributions but could we change this to "The authors" or "The contributors" or remove it altogether if it's optional?

**Review by tgeoghegan — CHANGES_REQUESTED** [review]

File: book.toml (hunk 2 above)
Comment (cjpatton) [reply book.toml:2]: Oops, that was added auto-magically

Comment (cjpatton) [reply book.toml]: Done.

**Review by tgeoghegan — APPROVED** [review]

**Pull request merged**